server/src/test/java/org/elasticsearch/common/settings/ConsistentSettingsIT.java

```diff
@@ -0,0 +1,289 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.settings;
+
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Setting.AffixSetting;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST)
+public class ConsistentSettingsIT extends ESIntegTestCase {
+
+    static final Setting<SecureString> DUMMY_STRING_CONSISTENT_SETTING = SecureSetting
+            .secureString("dummy.consistent.secure.string.setting", null, Setting.Property.Consistent);
+    static final AffixSetting<SecureString> DUMMY_AFIX_STRING_CONSISTENT_SETTING = Setting.affixKeySetting(
+            "dummy.consistent.secure.string.afix.setting.", "suffix",
+            key -> SecureSetting.secureString(key, null, Setting.Property.Consistent));
+    static final Setting<SecureString> EXTRA_DUMMY_STRING_CONSISTENT_SETTING = SecureSetting
+            .secureString("extra.dummy.consistent.secure.string.setting", null, Setting.Property.Consistent);
+    static final AffixSetting<SecureString> DUMMY_AFIX_STRING_INCONSISTENT_SETTING = Setting.affixKeySetting(
+            "dummy.inconsistent.secure.string.afix.setting.", "suffix",
+            key -> SecureSetting.secureString(key, null));
+    private final AtomicReference<Function<Integer, Settings>> nodeSettingsOverride = new AtomicReference<>(null);
+
+    public void testAllConsistentSuccess() throws Exception {
+        internalCluster().getInstances(Environment.class).forEach(environment -> {
+            ClusterService clusterService = internalCluster().getInstance(ClusterService.class);
+            assertTrue(new ConsistentSettingsService(environment.settings(), clusterService, Collections.emptyList()).areAllConsistent());
+            assertTrue(new ConsistentSettingsService(environment.settings(), clusterService,
+                    Collections.singletonList(DUMMY_STRING_CONSISTENT_SETTING)).areAllConsistent());
+            assertTrue(new ConsistentSettingsService(environment.settings(), clusterService,
```

Comment:
All of these assert true/false are very opaque. Could we have a helper assertion method that checks the consistency, and dumps the service state if it fails?

Reply:
If by opaque you mean hard to troubleshoot when they go awry, then there should be WARN logs on the node that failed the verification, that show the setting name and the hash value. I think those are enough? Do you see a need to publish the hashes in the junit output too? I think these are pretty useless anyway...

Reply:
Since you have been working on this, I'm sure you know exactly where to look in the logs. But some random dev that sees one of these tests fail has no idea when a true/false assertion fails with zero context in the test failure.

Reply:
I have added messages to assertions which also prints the hashes map.